### PR TITLE
JCRVLT-345 dedicated goals for validation

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/jackrabbit/filevault/maven/packaging/AbstractValidateMojo.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+     * <p>
+     * The repository-init feature of sling-start can define initial content that will be available in the
+     * repository before the first package is installed. Packages that depend on those nodes have no way to reference
+     * any dependency package that provides these nodes. A "real" package that would creates those nodes cannot be
+     * installed in the repository, because it would void the repository init structure. On the other hand would filevault
+     * complain, if the package was listed as dependency but not installed in the repository. So therefor this
+     * repository-structure packages serve as indicator packages that helps satisfy the structural dependencies, but are
+     * not added as real dependencies to the package.
+     */
+    @Parameter(property = "vault.reposistory.structure.packages")
```
Comment (adamcin) [inline src/main/java/org/apache/jackrabbit/filevault/maven/packaging/AbstractValidateMojo.java]: Spelling of `reposistory`.

**Comment by adamcin** [comment]
@kwin have you considered moving the validators SPI into the jackrabbit-filevault API instead? In this PR, you are tightly coupling it to maven, and it is a bit awkward for SPI implementors to add a maven plugin as a dependency in order to implement an API for publishing as a jar.

**Comment by kwin** [comment]
The use case is definitely to validate as part of the build (with a dedicated goal). So far I don't know of any other use case where you would want to leverage the validators, therefore I think the SPI naturally belongs to the Maven Plugin only. Maybe a dedicated SPI Maven Module (plain JAR) would be an option, but I would rather host it and release it together with the Maven Plugin. This would require some refactoring efforts, though. Do you have a use case in mind, where the validators should be used outside of the Maven context? Which registration mechanism would then be used (in case Sisu is not available)?

**Comment by adamcin** [comment]
@kwin well, I avoided using a specific DI facility in oakpal to reduce `oakpal-core` dependencies, and instead established a class loading behavior using the old, `get default constructor` style using reflection. All that is needed by the API is a classloader, so the `oakpal-maven-plugin` can provide one based on its `test`-scope classpath, while `oakpal-cli` can provide one using an `opear` file's `Bundle-ClassPath` manifest header. To publish a third-party library, you only need to depend on `oakpal-core`, implement `ProgressCheck` or `ProgressCheckFactory`, and reference the fully-qualified class name in an appropriate json file.

**Comment by kwin** [comment]
To get independent of Maven's DI one could also leverage https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html (which is IMHO the better fit). But again, before going in this direction I would like to hear, whether you have any use case in mind leveraging the validators outside the Maven Plugin?

**Comment by adamcin** [comment]
Projects that prefer `gradle`, or developers who actually use `vlt`?

**Comment by kwin** [comment]
Let's take this discussion to the mailing list https://lists.apache.org/list.html?dev@jackrabbit.apache.org

**Comment by kwin** [comment]
@adamcin I took a stab at that and it turned out to be easier than I thought. Please have a look at https://github.com/apache/jackrabbit-filevault/pull/58.

**Comment by kwin** [comment]
Closed via https://svn.apache.org/r1868139.

**Pull request closed without merging**